src/components/carousel/carousel.test.js

```diff
@@ -0,0 +1,229 @@
+
+        clickCarouselButton("left");
+
+        slide(indexSlide).should("have.text", title);
+      }
+    );
+
+    it.each([
+      ["0", "1", "Full clickable slide"],
+      ["1", "2", "Slide 3"],
+      ["2", "3", "Slide 4"],
+      ["3", "4", "Slide 5"],
+    ])(
+      "when right arrow is clicked from slide %s to slide %s title should be %s",
+      (index, indexSlide, title) => {
+        CypressMountWithProviders(<CarouselComponent />);
+
+        clickCarouselButton("right");
+
+        slide(indexSlide).should("have.text", title);
+      }
+    );
+
+    it("should verify left arrow button is disabled on first slide", () => {
+      CypressMountWithProviders(<CarouselComponent slideIndex={0} />);
+
+      previousArrowButton().should("be.disabled").and("have.attr", "disabled");
+    });
+
+    it("should verify right arrow button is disabled on last slide", () => {
+      CypressMountWithProviders(<CarouselComponent slideIndex={4} />);
+
+      nextArrowButton().should("be.disabled").and("have.attr", "disabled");
+    });
+
+    it.each([[true], [false]])(
+      "should verify slide Selector is enabled/disabled",
+      (isEnabled) => {
+        CypressMountWithProviders(
+          <CarouselComponent enableSlideSelector={isEnabled} />
+        );
+
+        if (isEnabled === false) {
+          slideSelector().should("not.exist");
+        } else {
+          slideSelector().should("exist");
+        }
+      }
+    );
+
+    it.each([[true], [false]])(
+      "should verify left arrow button is enabled/disabled",
+      (isEnabled) => {
+        CypressMountWithProviders(
+          <CarouselComponent enablePreviousButton={isEnabled} slideIndex={2} />
+        );
+
+        if (isEnabled === false) {
+          previousArrowButton().should("not.exist");
+        } else {
+          previousArrowButton()
+            .should("have.attr", "type", "button")
+            .and("not.have.attr", "disabled-type", "button");
+        }
+      }
+    );
+
+    it.each([[true], [false]])(
+      "should verify right arrow button is enabled/disabled %s",
+      (isEnabled) => {
+        CypressMountWithProviders(
+          <CarouselComponent enableNextButton={isEnabled} />
+        );
+
+        if (isEnabled === false) {
+          nextArrowButton().should("not.exist");
+        } else {
+          nextArrowButton()
+            .should("have.attr", "type", "button")
+            .and("not.have.attr", "disabled-type", "button");
+        }
+      }
+    );
+
+    it.each([
+      ["0", "Slide 1"],
+      ["1", "Full clickable slide"],
+      ["2", "Slide 3"],
+      ["3", "Slide 4"],
+      ["4", "Slide 5"],
+    ])("should set initial slide to %s", (index, title) => {
+      CypressMountWithProviders(
+        <CarouselComponent initialSlideIndex={index} />
+      );
+
+      slide(index).should("have.text", title);
+    });
+
+    it("should call click action when clicked", () => {
```

Comment:
This should read "should call onSlideChange when clicked". There should be an additional test case to test the `click` action. I would have expected this slide to be keyboard accessible too, in which case a `keydown` test would have been good to add, however it seems that functionality is not present as yet so just the `click` to add.